src/qt/peertablemodel.cpp

```diff
@@ -223,35 +131,52 @@ Qt::ItemFlags PeerTableModel::flags(const QModelIndex &index) const
     return retval;
 }
 
-QModelIndex PeerTableModel::index(int row, int column, const QModelIndex &parent) const
+QModelIndex PeerTableModel::index(int row, int column, const QModelIndex& parent) const
 {
     Q_UNUSED(parent);
-    CNodeCombinedStats *data = priv->index(row);
 
-    if (data)
-        return createIndex(row, column, data);
+    if (0 <= row && row < rowCount() && 0 <= column && column < columnCount()) {
+        return createIndex(row, column, const_cast<CNodeCombinedStats*>(&m_peers_data[row]));
+    }
+
     return QModelIndex();
 }
 
 void PeerTableModel::refresh()
 {
-    Q_EMIT layoutAboutToBeChanged();
-    priv->refreshPeers(m_node);
-    Q_EMIT layoutChanged();
-}
+    interfaces::Node::NodesStats nodes_stats;
+    m_node.getNodesStats(nodes_stats);
+    decltype(m_peers_data) new_peers_data;
+    new_peers_data.reserve(nodes_stats.size());
+    for (const auto& node_stats : nodes_stats) {
```

Comment:
Ugh, my suggestion was not good enought, you can actually move values out of that local `nodes_stats`:
```
    for (auto&& node_stats : nodes_stats) {
        const CNodeCombinedStats stats{std::get<0>(std::move(node_stats)), 
                                       std::get<2>(std::move(node_stats)),
                                       std::get<1>(std::move(node_stats))};
```
(note loop variable is no longer cost reference!)

Or even with an algorithm (also moves, at least it does on local test), if one would care about "No raw loops" mantra :) 
```
    std::transform(nodes_stats.begin(), nodes_stats.end(), std::back_inserter(new_peers_data), [](auto&& i) {
        return CNodeCombinedStats{std::get<0>(std::move(i)), std::get<2>(std::move(i)), std::get<1>(std::move(i))};
    });

```
Though moving `std::transform` seems fishy, not sure if it's "legal" to move source... But gcc 10.2.1 does that.

Reply:
> Ugh, my suggestion was not good enought...

It seems like a kind of premature optimization, no?

Reply:
Maybe, probably. Since there will be mostly on up to ~128 nodes by default, "nobody" would care if there are some extra copies I guess, and `std::transform` variant do look particularly "noisy" (i.e. "optimization" not worth the effort/readability cost).